distributed/worker.py

```diff
@@ -2350,7 +2352,7 @@ async def get_profile(
         if stop is None:
             istop = None
         else:
-            istop = bisect.bisect_right(history, (stop,)) + 1
+            istop = bisect.bisect_right(history, (stop,))
```

Comment:
This is an off by one error. Hard to find and very timing sensitive. The tests are not necessarily sensitive to it